File: UI/src/main/java/ca/warp7/rt/java/event/EventFeature.java
package ca.warp7.rt.java.event;

import ca.warp7.rt.java.core.ft.Feature;
import ca.warp7.rt.java.core.ft.FeatureItemTab;
import javafx.collections.FXCollections;

import java.util.List;

public class EventFeature implements Feature {

    @Override
    public List<FeatureItemTab> getLoadedTabs() {
        return FXCollections.singletonObservableList(new FeatureItemTab(
                "Event Overview", "fas-trophy:16:gray", getFeatureId(),
                FeatureItemTab.Group.SingleTab, ""
        ));
    }

    @Override
    public String getFeatureId() {
        return "event";
    }
}

File: UI/src/main/kotlin/ca/warp7/rt/ext/event/EventFeature.kt
package ca.warp7.rt.ext.event

import ca.warp7.rt.core.feature.Feature
import ca.warp7.rt.core.feature.FeatureItemTab
import ca.warp7.rt.core.feature.FeatureLink

class EventFeature : Feature {
    override val link get() = FeatureLink("Event Overview", "fas-trophy", 16)

    override val loadedTabs
        get() = listOf(FeatureItemTab(
                "Event Overview", "fas-trophy:16:gray", featureId,
                FeatureItemTab.Group.SingleTab, ""
        ))

    override val featureId get() = "dataset"
}
